lib/Parse/ParseExpr.cpp

```diff
@@ -2905,9 +2913,49 @@ ExprResult Parser::ParseBoundsExpression() {
     SkipUntil(tok::r_paren, StopAtSemi | StopBeforeMatch);
 
   PT.consumeClose();
+
+  if (StartsRelativeBoundsClause(Tok)) {
```

Comment:
Ah, yes, this is the correct place.   The Github pull request omits lines of code and I got confused about where this had been placed.

I think this if-statement should only be executed only if the code started parsing a range bounds expression before.  You may need to add a flag to record this.